community/values/src/main/java/org/neo4j/values/NoValue.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values;
+
+final class NoValue extends Value
+{
+    @SuppressWarnings( "WeakerAccess" )
+    public static NoValue NO_VALUE = new NoValue();
+
+    private NoValue()
+    {
+    }
+
+    @Override
+    public boolean equals( Object other )
+    {
+        return false;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return 123456789;
```

Comment:
why this hash code?
in order to avoid other people wondering that very same thing in the future, replace it with `System.identityHashCode( this )` (which is what the default `Object.hashCode()` implementation returns)